renter/formcontract.go

```diff
@@ -0,0 +1,153 @@
+package renter
+
+import (
+	"errors"
+	"fmt"
+
+	"go.sia.tech/core/net/rhp"
+	"go.sia.tech/core/net/rpc"
+	"go.sia.tech/core/types"
+)
+
+// FormContract negotiates a new contract with the host using the specified
+// funds and duration.
+func (s *Session) FormContract(renterKey types.PrivateKey, hostFunds, renterFunds types.Currency, endHeight uint64) (types.FileContractRevision, []types.Transaction, error) {
```

Comment:
I think this function should just return a `FileContract`, i.e. the `Revision` field of what it's returning now. Having a whole `FileContractRevision` isn't really useful to the renter at this stage. (And it won't be valid anyway, since it doesn't have an accumulator proof.)

Reply:
I chose to return a `FileContractRevision` to pass the signatures to the caller. Should we consider moving the signatures directly to the `FileContract`?